connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/MsSqlServerIntegrationSuite.scala

```diff
@@ -146,4 +146,70 @@ class MsSqlServerIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JD
         |""".stripMargin)
     assert(df.collect().length == 2)
   }
+
+  test("SPARK-50087: SqlServer handle booleans in IF in SELECT test") {
```

Comment:
does this provide extra test coverage than the following new test cases?

Reply:
This is a specific edge case. If you use a single IF in a SELECT it won't be pushed down, but this is.